[datadog_integration_azure] Add app_service_plan_filters, cspm, and custom metrics (#2059)
DataDog/terraform-provider-datadog: docs/resources/integration_azure.md

### Optional

> - `app_service_plan_filters` (String) String of app service plan tag(s) (in the form `key:value,key:value`) defines a filter that Datadog will use when collecting metrics from Azure. Limit the Azure instances that are pulled into Datadog by using tags. Only hosts that match one of the defined tags are imported into Datadog. e.x. `env:production,deploymentgroup:red`

Reviewer: ```suggestion
- `app_service_plan_filters` (String) String of app service plan tag(s) (in the form `key:value,key:value`) defines a filter that Datadog uses when collecting metrics from Azure. Limit the Azure instances that are pulled into Datadog by using tags. Only hosts that match one of the defined tags are imported into Datadog. For example, `env:production,deploymentgroup:red`.
```
